lib/config/fileconf.go

```diff
@@ -609,6 +615,32 @@ func checkAndSetDefaultsForAzureMatchers(matcherInput []AzureMatcher) error {
 	return nil
 }
 
+func checkAndSetDefaultsForAzureInstaller(matcher *AzureMatcher) error {
+	if matcher.InstallParams == nil {
+		matcher.InstallParams = &InstallParams{
+			JoinParams: JoinParams{
+				TokenName: defaults.AzureInviteTokenName,
+				Method:    types.JoinMethodAzure,
+			},
+			ScriptName: installers.InstallerScriptName,
+		}
+	} else {
+		if method := matcher.InstallParams.JoinParams.Method; method == "" {
+			matcher.InstallParams.JoinParams.Method = types.JoinMethodAzure
+		} else if method != types.JoinMethodAzure {
+			return trace.BadParameter("only Azure joining is supported for Azure auto-discovery")
+		}
+		if token := matcher.InstallParams.JoinParams.TokenName; token == "" {
```

Comment:
Suggestion: add some spacing, conditions seem to treat different "steps" of validation.

```suggestion
		}

		if token := matcher.InstallParams.JoinParams.TokenName; token == "" {
```